### PR TITLE
Tree shake workbox

### PR DESCRIPTION
**Pull request by prateekbh**
- Followup after #1 
- Builds a custom build of Workbox with rollup
- Removes tempdir dependency
- Adds closure compiler for minification
- Final build size of module + workbox is 2.7kb 🤯🤯🤯

rdj_woah

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
Perhaps make the merge target `amp-caching`? That would lower the diff a bit :)

File: src/modules/amp-caching/index.ts [hunk 1]
```diff
@@ -1,3 +1,29 @@
+// @ts-ignore
```
Comment (kristoferbaxter) [inline src/modules/amp-caching/index.ts]: TIL: This is going to save me some time.

File: src/modules/amp-caching/index.ts [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+import router from 'workbox-routing';
+// @ts-ignore
+import { CacheFirst, StaleWhileRevalidate } from 'workbox-strategies';
+
 export function ampAssetsCaching() {
+  const CACHE_OPTIONS = {
```
Comment (kristoferbaxter) [inline src/modules/amp-caching/index.ts]: Since this is static, does it make sense to hoist outside of the function?

**Comment by prateekbh** [comment]
Lets get #1 in first cuz thats functional and we have tests to verify the changes. That'll lower the diff

File: src/modules/amp-caching/index.ts (hunk 2 above)
Comment (prateekbh) [reply src/modules/amp-caching/index.ts]: done

**Comment by prateekbh** [comment]
removed workbox core, now is only 2682 bytes

**Comment by prateekbh** [comment]
Merging as this became base for multiple branches, please still feel free to comment on stuff and i'll carry it over in next PRs

File: package.json [hunk 3]
```diff
@@ -33,14 +34,17 @@
     "promisify": "0.0.3",
     "puppeteer": "1.9.0",
     "rollup": "0.66.6",
+    "rollup-plugin-node-resolve": "^3.4.0",
```
Comment (kristoferbaxter) [inline package.json]: Let's pin these dependencies and use renovate to manage updates.

**Pull request merged**